packages/audiofileplayer/android/src/main/java/com/google/flutter/plugins/audiofileplayer/AudiofileplayerPlugin.java

```diff
@@ -112,37 +121,103 @@
   // Used when defining an Intent from a custom media button.
   public static final String CUSTOM_MEDIA_BUTTON_EXTRA_KEY = "customMediaButton";
 
-  private final Registrar registrar;
-  private final Map<String, ManagedMediaPlayer> mediaPlayers;
-  private final MethodChannel methodChannel;
+  private Registrar registrar;
+  private Activity activity;
+  private Map<String, ManagedMediaPlayer> mediaPlayers;
+  private MethodChannel methodChannel;
+  private Context context;
+  private ActivityPluginBinding activityPluginBinding;
 
   private MediaBrowserCompat mediaBrowser;
   private MediaControllerCompat mediaController;
 
   public static void registerWith(Registrar registrar) {
-    final MethodChannel methodChannel = new MethodChannel(registrar.messenger(), CHANNEL);
-    final AudiofileplayerPlugin instance = new AudiofileplayerPlugin(registrar, methodChannel);
-    methodChannel.setMethodCallHandler(instance);
+    AudiofileplayerPlugin instance = new AudiofileplayerPlugin();
+    instance.registrar = registrar;
+    instance.activity = registrar.activity();
+    instance.initInstance(registrar.messenger(), registrar.context());
+    instance.registerLifecycleCallbacks(instance.activity);
```

Comment:
How do you handle this for v2 embedding?

I don't see lifecycle callbacks being registered onAttachedToActivity.

Reply:
Do we intent to keep the lifecycle callbacks for v2? I used attachToActivity() to connect the mediaBrowser like the one we have within lifecyclecallbacks.

Reply:
My understanding is that ActivityLifecycleCallbacks are different. They are handling activity lifecycles, onCreate, onStart, OnResume etc. The ActivityAware is simply for attaching/detaching activities to engine. You'll need to register the activity lifecycle callbacks for v2.